src/shared/pretty-print.c

```diff
@@ -160,6 +160,25 @@ static int cat_file(const char *filename, bool newline) {
                 if (r == 0)
                         break;
 
+                if (flags & CAT_TLDR) {
+                        const char *t = skip_leading_chars(line, WHITESPACE);
+
+                        if ((flags & CAT_FORMAT_HAS_SECTIONS) && *t == '[') {
+                                /* The start of a section, let's not print it yet. */
+                                free_and_replace(section, line);
+                                continue;
+                        }
+
+                        if (IN_SET(*t, '#', ';', '\0'))
```

Comment:
is this correct btw? i don#t think it#s that easy, because some of our config files allow '#' as comments *and* ';' too, but others only the former.

bootspec, crypttab, verittab only do '#', afaics...

Reply:
`cat-config` cannot be used to show bootspec entries. `bootctl` has it's own display helper.
In principle `cat-config` could be used to show `fstab`/`crypttab`/`veritytab`, but not very well… Those formats don't support drop-ins or the whole usr-etc-run hierarchy. For the formats that would be shown with `cat-config` or one of the `--cat-config`/`--tldr` switches, we always support both `#` and `;`.

So I don't think this needs to be handled now. If somebody ever extends those helpers to support that they'll need to figure this out, but it'll most likely require having a list of specific file names which use a custom format…